src/lib/gundb/FeedStorage.js

```diff
@@ -980,7 +980,7 @@ export class FeedStorage {
 
         // if no item in the cache and it's some transaction
         // then getting tx item details from the wallet
-        if (!item && id.startsWith('0x')) {
+        if ((!item && id.startsWith('0x')) || (item && !get(item.data.reason))) {
```

Comment:
for some reason (maybe a race condition) when trying to getFromOutbox when just receiving the tx the data is null
when getting the feed page, if the reason is missing, forcing to recheck the receipt enables to find the outbox data

Reply:
ok that could be, but we do not need to parse the whole receipt every time.
we need to read the inbox only in case of a TxType.RECEIVE and then we to only call the getfrominbox